pkgs/development/interpreters/gnudatalanguage/default.nix

```diff
@@ -0,0 +1,188 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, wrapGAppsHook
+, readline
+, ncurses
+, zlib
+, gsl
+, openmp
+, graphicsmagick
+, fftw
+, fftwFloat
+, fftwLongDouble
+, proj
+, shapelib
+, expat
+, udunits
+, eigen
+, pslib
+, eccodes
+, glpk
+, libpng
+, plplot
+, libtiff
+, libgeotiff
+, libjpeg
+  # We enable it in hdf4 and use libtirpc as a dependency here from the passthru
+  # of hdf4
+, enableLibtirpc ? stdenv.isLinux
+, libtirpc
+, python3
+, enableMPI ? (stdenv.isLinux || stdenv.isDarwin)
+  # Choose MPICH over OpenMPI because it currently builds on AArch and Darwin
+, mpi
+  # Unfree optional dependency for hdf4 and hdf5
+, enableSzip ? false
+, szip
+, enableHDF4 ? true
+, hdf4
+  # netcdf with mpiSupport=true
+  # depends on hdf5 with usev110Api=true
+  # If you want them all,
+  # uncouple the hdf5 used by gnudatalanguage and that used by netcdf
+  # by overriding `hdf5-forced` and `netcdf-forced`.
+  # Do so at your own risk.
```

Comment:
Why is this risky? If it's not risky but you wish to not support it officially in the expression, perhaps it'd be nice to make it easy to do so with a new flag:

```nix
# A comment explaining it's not supported or whatever
, enableNetcdfMpi ? false
```